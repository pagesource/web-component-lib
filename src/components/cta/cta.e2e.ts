import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('cta E2E Tests >>>', () => {
  it('should render a cta Component', async () => {
    const page: E2EPage = await newE2EPage({
      html: `<wc-cta></wc-cta>`
    });

    const results = await page.compareScreenshot();
    // expect(results).toMatchScreenshot();
  });
});
