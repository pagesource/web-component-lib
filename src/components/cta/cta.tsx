/**
 *
 * cta
 *
 */

import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: `wc-cta`,
  styleUrls: {
    default: './cta.default.scss'
  },
  shadow: true
})
export class CtaComponent {
  /**
   * Optional. Specify which of your styleUrls to apply.
   * Defaults to the value of the global Context.mode defined in app.ts.
   * Each "mode" results in a separate component dist file, thus avoiding unused css in component variants.
   */
  @Prop() public mode?: string;

  // optional message prop
  @Prop() public displaytext?: string;
  /**
   * Set href attribute if you want the output to be link else button will be created
   */
  @Prop() public href?: string;

  public render() {
    let props = {};
    /**
     * Choose if href is passed in tag then a tag is generated else button tag is generated
     */
    const Wrapper: any = this.href ? 'a' : 'button';
    /**
     * Set href props
     */

    if (this.href) {
      props = {
        href: this.href
      };
    }
    return (
      <Wrapper {...props}>
        <slot></slot>
        {this.displaytext}
      </Wrapper>
    );
  }
}
