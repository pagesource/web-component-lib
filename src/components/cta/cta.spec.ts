import { newSpecPage } from '@stencil/core/dist/testing';

import { CtaComponent } from './cta';

const compMarkup = `<wc-cta></wc-cta>`;

describe('cta Spec Tests >>>', () => {
  it('should render a cta Component', async () => {
    const page = await newSpecPage({
      html: compMarkup,
      components: [CtaComponent],
      supportsShadowDom: false
    });
    expect(page.root).not.toBeNull();
    // snapshot testing
    expect(page.root).toMatchSnapshot();
  });
});
