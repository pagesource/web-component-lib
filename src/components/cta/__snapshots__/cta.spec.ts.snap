// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cta Spec Tests >>> should render a cta Component 1`] = `
<wc-cta>
  <!---->
  <button></button>
</wc-cta>
`;
